components/automate-deployment/pkg/backup/context.go

```diff
@@ -101,6 +101,13 @@ func WithContextBackupLocationSpecification(locationSpec LocationSpecification)
 	}
 }
 
+func WithContextDeadline(deadline time.Time) ContextOpt {
+	return func(c *Context) {
+		newCtx := context.WithValue(c.ctx, "deadline", deadline)
```

Comment:
where are you using this stored deadline value? We have to create the child contexts with this deadline value on each and every backup process.